File: 0204-count-primes/0204-count-primes.java
class Solution {
    public int countPrimes(int n) {
        if(n == 0 || n == 1)return 0;
        int[] prime = new int[n + 1];//+1 is to considering zero too

        for(int i = 2; i < Math.sqrt(n);i++){
            if(prime[i] == 0){
                for(int j = i;i*j < n;j++){
                    prime[i * j] = 1;
                }
            }
        }
        int res = 0;
        for(int i = 2;i < n;i++){
            if(prime[i] == 0)res++;
        }
        return res;
    }
}

File: 0204-count-primes/0204-count-primes.py
class Solution:
    def countPrimes(self, n: int) -> int:
        if n < 2:
            return 0
    
        prime = [0] * n  # 0 means prime
        for i in range(2, int(n**0.5) + 1):
            if prime[i] == 0:
                for j in range(i * i, n, i):  # Start from i*i
                    prime[j] = 1
        return sum(1 for i in range(2, n) if prime[i] == 0)